web/src/pages.rs

```diff
@@ -146,3 +151,68 @@ pub async fn my_articles(
     })
     .into_response())
 }
+
+#[axum::debug_handler(state = AppState)]
+pub async fn my_categories(
+    t: Template,
+    State(state): State<AppState>,
+    user: User,
+) -> WebResult<Response> {
+    info!("Fetching categories for user_id: {}", user.user_id);
+
+    let categories = sqlx::query!("SELECT * FROM category",)
+        .fetch_all(&state.db)
+        .await?;
+
+    Ok(t.render(maud::html! {
+        h1 { "My Categories" }
+        ul {
+            @for category in categories {
+                @let category_name = category.category.unwrap_or("No category".to_string());
+                li {
+                    a href=(format!("/categories/{}", category.category_id)) { (category_name) }
+                }
+            }
+        }
+    })
+    .into_response())
+}
+
+#[axum::debug_handler(state = AppState)]
+pub async fn articles_by_category(
+    t: Template,
+    Path(category_id): Path<Uuid>,
+    State(state): State<AppState>,
+    user: User,
+) -> WebResult<Response> {
+    info!("Fetching articles for category_id: {}", category_id);
+
+    let articles = sqlx::query!(
```

Comment:
In this query we connect related data across multiple tables `category`, `markdown`, `pagesnapshot`, and `pages` to retrieve all page information.